src/catalog/catalog_entry/catalog_entry.cpp

```diff
@@ -35,5 +35,12 @@ std::unique_ptr<CatalogEntry> CatalogEntry::deserialize(common::Deserializer& de
     return entry;
 }
 
+void CatalogEntry::copy(CatalogEntry& other) const {
```

Comment:
i would suggest renaming it to copyFrom. Instead of copying fields to `other`, we can let `other` copy from the current catalogEntry.